src/Collector/Twig/HttpMessageMarkupExtension.php

```diff
@@ -34,40 +34,14 @@ public function __construct(?ClonerInterface $cloner = null, ?DataDumperInterfac
     public function getFilters()
     {
         return [
-            new TwigFilter('httplug_markup', $this->markup(...), ['is_safe' => ['html']]),
             new TwigFilter('httplug_markup_body', $this->markupBody(...), ['is_safe' => ['html']]),
         ];
     }
 
-    /**
-     * @param string $message http message
-     */
-    public function markup($message)
-    {
-        @trigger_error('"httplug_markup" twig extension is deprecated since version 1.17 and will be removed in 2.0. Use "@Httplug/http_message.html.twig" template instead.', E_USER_DEPRECATED);
-
-        $safeMessage = htmlentities($message);
-        $parts = preg_split('|\\r?\\n\\r?\\n|', $safeMessage, 2);
-
-        if (!isset($parts[1])) {
-            // This is not a HTTP message
-            return $safeMessage;
-        }
-
-        if (empty($parts[1])) {
-            $parts[1] = '(This message has no captured body)';
-        }
-
-        // make header names bold
-        $headers = preg_replace("|\n(.*?): |si", "\n<b>$1</b>: ", $parts[0]);
-
-        return sprintf("%s\n\n<div class='httplug-http-body httplug-hidden'>%s</div>", $headers, $parts[1]);
-    }
-
     public function markupBody(string $body): ?string
     {
-        if (in_array(substr($body, 0, 1), ['{', '['], true)) {
-            $json = json_decode($body, true);
+        if ('' !== $body && in_array($body[0], ['{', '['], true)) {
```

Comment:
str_starts_with does not accept an array. i am not changing the scope, i only verify that the body is not empty, because then we would try to access an array out of its bounds. (which probably has been prevented somewhere else, or we would already have seen issues). but in the interest of defensive code i think we should not risk accessing the string outside its bounds.